dds/DCPS/DataReaderImpl_T.h

```diff
@@ -897,6 +897,67 @@ namespace OpenDDS {
     return inst;
   }
 
+  DDS::ReturnCode_t setup_deserialization()
```

Comment:
I'd like to to avoid adding more functions to headers when possible in the long run because they slow down compilation. Instead you could add a virtual method to `DataReaderImpl` to get the max extensibility.